controlplane/kubeadm/controllers/controller.go

```diff
@@ -421,32 +421,29 @@ func (r *KubeadmControlPlaneReconciler) ClusterToKubeadmControlPlane(o handler.M
 // It removes any etcd members that do not have a corresponding node.
 // Also, as a final step, checks if there is any machines that is being deleted.
 func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, controlPlane *internal.ControlPlane) (ctrl.Result, error) {
-	logger := controlPlane.Logger()
 
 	// Do a health check of the Control Plane components
 	if err := r.managementCluster.TargetClusterControlPlaneIsHealthy(ctx, util.ObjectKey(cluster)); err != nil {
-		logger.V(2).Info("Waiting for control plane to pass control plane health check to continue reconciliation", "cause", err)
 		r.recorder.Eventf(kcp, corev1.EventTypeWarning, "ControlPlaneUnhealthy",
 			"Waiting for control plane to pass control plane health check to continue reconciliation: %v", err)
-		return ctrl.Result{RequeueAfter: healthCheckFailedRequeueAfter}, nil
+		return ctrl.Result{RequeueAfter: healthCheckFailedRequeueAfter}, errors.Wrap(err, "Waiting for control plane to pass control plane health check to continue reconciliation")
```

Comment:
If we return an error, `ctrl.Result` is ignored

Reply:
Hmm. Am I reading this correctly: https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/scale.go#L67

Looks like `result` is returned and we probably want to keep `healthCheckFailedRequeueAfter` as the retry interval instead of just returning `ctrl.Result{}`

Reply:
Yes both are returned, although when the reconciler returns them to controller-runtime's controller, the code looks only at the error, if one is present. See https://github.com/kubernetes-sigs/controller-runtime/blob/e50c7b819263665e0406317a73e4e1c72f2fe2fb/pkg/internal/controller/controller.go#L240-L243


If we want to use a fixed requeue, return just `ctrl.Result` and the log the error, if we want to make the controller go in exponential backoff loop instead, we can just return the error

Reply:
Ah okay. 
I think there was a strong desire to reduce the number of logging statements in the code. I'll change it to return error.